srlife/damage.py

```diff
@@ -517,6 +729,50 @@ def calculate_eq_stress(self, mandel_stress, temperatures, material):
         # Projected equivalent stress
         return 0.5 * (sigma_n + np.sqrt((sigma_n**2) + (tau**2)))
 
+    def calculate_kbar(self, temperatures, material, A, dalpha, dbeta):
+        """
+        Calculate the evaluating normalized crack density coefficient (kbar)
+        using the Weibull scale parameter (svals), Weibull modulus (mvals),
+        poisson ratio (nu)
+        """
+
+        self.temperatures = temperatures
+        self.material = material
+
+        # Weibull modulus
+        mvals = self.material.modulus(temperatures)
+
+        # Temperature average values
+        mavg = np.mean(mvals, axis=0)
+
+        # Material parameters
+        nu = np.mean(material.nu(temperatures), axis=0)
+
+        # Evaluating integral for kbar while suppressing warning given when
+        # negative numbers are raised to rational numbers
+        with np.errstate(invalid="ignore"):
```

Comment:
I don't like the `errstate` stuff, or at least I want to understand why it's necessary.

Reply:
Removed all the errstates and ensured no error is shown.